qcodes/utils/measurement_subscribers.py

```diff
@@ -0,0 +1,93 @@
+"""
+This module contains example functions and classes that can subscribe to a
+``Measurement`` instance with ``add_subscriber``.
+"""
+
+import time
+import datetime
+from typing import (Optional, List, Tuple, Any, Union, MutableSequence,
+                    MutableMapping)
+
+
+class TextProgress:
```

Comment:
by looking at this code i strongly recommend to dig deeper into `tqdm` and instead of reinventing it - extend it when needed (which i don't think that extending is needed per se). this is how to extend tqdm - https://github.com/tqdm/tqdm#description-and-additional-stats , this is about nested progress bars - https://github.com/tqdm/tqdm#nested-progress-bars (their support has reasonable constraints), and jupyter integration is awesome - https://github.com/tqdm/tqdm#ipythonjupyter-integration .

and perhaps this little post will help finding a good way to display progress of the measurement - https://github.com/tqdm/tqdm/wiki/How-to-make-a-great-Progress-Bar

As a bonus, qcodes already has tqdm as a dependency :)